lib/DocumentRetriever.ts

```diff
@@ -332,15 +305,17 @@ DocumentRetriever.prototype.all = function (this: DocumentRetriever, delay = 0,
 	return this;
 };
 
+export interface Scan<T> extends DocumentRetriever, BasicOperators<Scan<T>> {
+	exec(): Promise<ScanResponse<T>>;
+	exec(callback: CallbackType<ScanResponse<T>, AWSError>): void;
+}
 
-export class Scan extends DocumentRetriever {
-	exec(): Promise<ScanResponse<Document[]>>;
-	exec(callback: CallbackType<ScanResponse<Document[]>, AWSError>): void;
-	exec (callback?: CallbackType<ScanResponse<Document[]>, AWSError>): Promise<ScanResponse<Document[]>> | void {
+export class Scan<T> extends DocumentRetriever {
```

Comment:
Is there a reason for having an interface and a class? Isn't there a way to combine these into 1 thing? Or?